Products/ZenHub/services/DiscoverService.py

```diff
@@ -303,3 +303,25 @@ def remote_moveDevice(self, dev, path):
     def remote_getDefaultNetworks(self):
         monitor = self.dmd.Monitors.Performance._getOb(self.instance)
         return [net for net in monitor.discoveryNetworks]
+
+    @translateError
+    def remote_removeInterfaces(self, net):
+        """
+        Remove IPs for particular network
+        already assigned to interfaces (device components)
+
+        @param net - network to discover
+        @return:  a list of IPs without addresses assigned for interfaces
+        @rtype: list
+        """
+
+        full_ip_list = net.fullIpList()
+
+        for d in self.dmd.Devices.getSubDevices():
```

Comment:
I suggest using the `getSubDevicesGen` method instead.  The `getSubDevices` method returns a list while `getSubDevicesGen` method returns a generator.  Since this is called at the root devices organizer, this will return all the devices, and `getSubDevices` would load all the devices into memory.  Using the `getSubDevicesGen` method will load the devices one at at a time as the results are iterated over which, hopefully, won't be as memory intensive (i.e. garbage collection of device objects that aren't needed anymore).

Reply:
added requested change